blog/_posts/2024-08-19-KAITO-finetuning.md

```diff
@@ -0,0 +1,139 @@
+---
+title: "Fine tune language models with KAITO on AKS"
+description: "Make your AI model even smarter, in just a few steps"
+date: 2024-08-23
+author: Sachi Desai
+categories: AI
+---
+
+## Introduction
+
+You may have heard of the [Kubernetes AI Toolchain Operator](https://github.com/Azure/kaito/tree/main) (KAITO) announced at Ignite 2023 and KubeCon Europe this year. The open source project has gained popularity in recent months by introducing a streamlined approach to AI model deployment and flexible infrastructure provisioning on Kubernetes.
+
+With the [v0.3.0 release](https://github.com/Azure/kaito/releases/tag/v0.3.0), KAITO has expanded the supported model library to include the Phi-3 model, but the biggest (and most exciting) addition is the ability to fine-tune open-source models. Why should you be excited about fine-tuning? Well, it’s because fine-tuning is one way of giving your foundation model additional training using a specific dataset to enhance accuracy, which ultimately improves the interaction with end-users.
+
+## Fine-tuning a foundation model is sometimes necessary
+
+Let’s experiment with [Phi-3-mini-128K](https://huggingface.co/microsoft/Phi-3-medium-128k-instruct), a robust and high-performing model that’s a bit smaller in size than your average LLM. We’ll use the following [chatbot UI tool](https://streamlit.io/), connected to a Phi-3 inference service deployed by KAITO, and ask a basic question about AKS:
+
+![Screenshot of untuned model response](/blog/assets/images/kaito_untuned_phi3_response.png)
+*Screenshot of un-tuned model response*
+
+Uh oh! That is **NOT** the answer we were expecting – we’d expect the model to know more about Kubernetes and cloud technologies. This foundation model was trained on a mix of synthetic data and public websites, so it doesn’t recognize AKS as Azure Kubernetes Service, and got creative in answering the question to the best of its ability. This is a well-known problem in the AI community and various fine-tuning techniques have been invented to enhance model accuracy using domain-specific knowledge. In our specific use case, we need the Phi-3 model to have more knowledge of Kubernetes and cloud platforms.
+
+## KAITO can help with model fine-tuning
+
+KAITO supports Low-Rank Adaption (LoRA), the most common parameter-efficient fine-tuning technique, which significantly lowers the GPU requirement for tuning a language model. With LoRA, fine-tuning an LLM in a single GPU is possible! 
```

Comment:
Naive reader questions:
* any link/literature to support this is the most common technique? 
* will we consider supporting others?